src/gui/src/displayControls.cpp

```diff
@@ -508,6 +559,46 @@ bool DisplayControls::isVisible(const odb::dbTechLayer* layer)
   return false;
 }
 
+bool DisplayControls::isInstanceVisible(odb::dbInst* inst)
+{
+  dbMaster* master = inst->getMaster();
+  if (master->isEndCap()) {
+    return instances_.endcap.visible->checkState() == Qt::Checked;
+  } else if (master->isFiller()) {
+    return instances_.fill.visible->checkState() == Qt::Checked;
+  } else if (master->isCore()) {
+    return instances_.core.visible->checkState() == Qt::Checked;
+  } else if (master->isBlock()) {
+    return instances_.blocks.visible->checkState() == Qt::Checked;
+  } else if (master->isPad()) {
+    return instances_.pads.visible->checkState() == Qt::Checked;
+  } else if (master->getType() == dbMasterType::COVER || master->getType() == dbMasterType::COVER_BUMP) {
+    return instances_.cover.visible->checkState() == Qt::Checked;
```

Comment:
I'm ok with you adding isCover() to OpenDB in the same style as the rest of these.  If you prefer not to I can add it.

Reply:
I added it let me know if it needs any changes.